File: 11720.java
import java.util.Scanner;

public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int length = sc.nextInt();                 //nextInt와 nextLine같이 사용시 nextInt에 입력할때의 \n이 nextLine에 입력되어버림.
        String num = sc.next();                    //공백없이 받으려면 next()가 맞음. nextLine은 \n까지를 입력으로 받아옴. 그래서 nextInt와 사용시 씹힘 문제가 발생함.
        sc.close();
        
        int sum = 0;
        for(int i = 0; length > i; i++)
            sum = sum + num.charAt(i) - '0';        //자바도 동일하게 형변환. charAt을 이용해 인덱스로 문자 추출
        
        System.out.println(sum);
    }
}

File: 11720.c
#include<stdio.h>

int main(){
    int size, sum = 0;
    char num[100] = {0, };                //초기화
    scanf("%d\n", &size);
    scanf("%s", num);                    //배열의 이름이 곧 주소이다.
    
    for(int i = 0; size > i; i++)        //사이즈는 5라도 인덱스는 0~4임. 5에는 \0이 들어감.
        sum = sum + num[i] - '0';
    
    printf("%d", sum);
    
    return 0;
}
